includes/course-theme/class-sensei-course-theme-lesson.php

```diff
@@ -142,4 +149,79 @@ private function maybe_add_prerequisite_notice() {
 			$notices->add_notice( 'locked_lesson', $text, __( 'You don\'t have access to this lesson', 'sensei-lms' ), [], 'lock' );
 		}
 	}
+
+	/**
+	 * Maybe add not enrolled notice.
+	 *
+	 * @return void
+	 */
+	private function maybe_add_not_enrolled_notice() {
+		$lesson_id = \Sensei_Utils::get_current_lesson();
+		$course_id = Sensei()->lesson->get_course_id( $lesson_id );
+		$notices   = \Sensei_Context_Notices::instance( 'course_theme_locked_lesson' );
+
+		if ( ! is_user_logged_in() ) {
+			$user_can_register = get_option( 'users_can_register' );
+
+			// Take course URL.
+			$take_course_redirect_url = add_query_arg( 'take_course_sign_in', '1', get_permalink( $course_id ) );
+			$take_course_url          = $user_can_register ? sensei_user_registration_url( true, $take_course_redirect_url ) : sensei_user_login_url( $take_course_redirect_url );
```

Comment:
As the default option, we're sending logged-out users to the registration form. If registration is not activated, we send them to the login form.